examples/hello-world/main.cpp

```diff
@@ -323,8 +322,9 @@ Slang::Result initialize()
     swapchainDesc.height = gWindowHeight;
     swapchainDesc.imageCount = kSwapchainImageCount;
     swapchainDesc.queue = gQueue;
-    gSwapchain = gRenderer->createSwapchain(
-        swapchainDesc, gfx::WindowHandle::FromHwnd(getPlatformWindowHandle(gWindow)));
+    gfx::WindowHandle windowHandle;
+    memcpy(&windowHandle, &gWindow->getNativeHandle(), sizeof(windowHandle));
```

Comment:
Should `getNativeHandle()` be something like:

```
Result getNativeHandle(size_t dataSize, void* outData);
```
?

It seems like we don't have a lot of protection against copying out the native handle expecting one kind and then getting another, but maybe that is unavoidable. I guess in theory we could have `gfx::WindowHandle` include a tag to identify the flavor of the handle it holds, and then the `platform` layer could specifically return a `gfx::WindowHandle` instead of something opaque.